client/incidents.go

```diff
@@ -112,41 +106,70 @@ func (s *IncidentsService) Create(ctx context.Context, opts IncidentCreateOption
 
 // Get an incident.
 func (s *IncidentsService) Get(ctx context.Context, incidentID string) (*Incident, error) {
-	u := fmt.Sprintf("%s/%s", "incidents", incidentID)
-	req, err := s.client.NewRequest(http.MethodGet, u, nil)
+	incidentURL := fmt.Sprintf("incidents/%s", incidentID)
+	req, err := s.client.NewRequest(http.MethodGet, incidentURL, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	i := new(Incident)
-	resp, err := s.client.Do(ctx, req, i)
+	incident := new(Incident)
+	resp, err := s.client.Do(ctx, req, incident)
 	if err != nil {
 		return nil, err
 	}
 	resp.Body.Close()
 
-	return i, nil
+	return incident, nil
 }
 
 // GetByChannelID gets an incident by ChannelID.
 func (s *IncidentsService) GetByChannelID(ctx context.Context, channelID string) (*Incident, error) {
-	return nil, errors.New("not implemented")
+	channelURL := fmt.Sprintf("incidents/channel/%s", channelID)
+	req, err := s.client.NewRequest(http.MethodGet, channelURL, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	incident := new(Incident)
+	resp, err := s.client.Do(ctx, req, incident)
+	if err != nil {
+		return nil, err
+	}
+	resp.Body.Close()
+
+	return incident, nil
 }
 
 // Update an incident.
-func (s *IncidentsService) Update(ctx context.Context, incidentID string, opts IncidentUpdateOptions) (*Incident, error) {
-	return nil, errors.New("not implemented")
+func (s *IncidentsService) Update(ctx context.Context, incidentID string, opts incident.UpdateOptions) (*Incident, error) {
+	incidentURL := fmt.Sprintf("incidents/%s", incidentID)
+	incidentUpdateRequest := Incident{
+		CommanderUserID: *opts.CommanderUserID,
+	}
+
+	req, err := s.client.NewRequest(http.MethodPatch, incidentURL, incidentUpdateRequest)
```

Comment:
We can directly pass the `opts` param as the last parameter in `NewRequest` (so we don't need to build an Incident with that info). We need to keep the pointers as received, even if they are nil.
```suggestion
	req, err := s.client.NewRequest(http.MethodPatch, incidentURL, opts)
```